tsdb/shard.go

```diff
@@ -909,6 +930,15 @@ func (f *FieldCodec) FieldByName(name string) *Field {
 	return f.fieldsByName[name]
 }
 
+func NewFieldKeysIterator(sh *Shard, opt influxql.IteratorOptions) (influxql.Iterator, error) {
+	fn := func(m *Measurement) []string {
+		keys := m.FieldNames()
+		sort.Strings(keys)
+		return keys
+	}
+	return newMeasurementKeysIterator(sh, fn, opt)
+}
+
```

Comment:
Mostly because I was copying verbatim large portions of functionality with exactly one line change. So I decided to change both of the iterators to use the same base since they both did the same thing.